CAIPs/caip-345.md

```diff
@@ -0,0 +1,186 @@
+        headers: {
+            "Content-Type": "application/json"
+        },
+        body: JSON.stringify(jsonRpc)
+    });
+} else {
+    // fallback to sending directly to wallet
+}
+```
+
+### Usage in CAIP-25
+
+The `caip345` property can be used in both `sessionProperties` and `scopedProperties`, depending on what scopes the methods are supported on.
+
+```ts
+// scoped or session properties
+type Caip25Properties = {
+    caip345: WalletService[],
+    [key: string]: any, // other properties
+};
+```
+
+If a method is included in the `caip345` property, then the same method MUST be included in the CAIP-25 session for the same scopes. Wallets MUST implement fallback handling for all wallet service methods, in case the app does not implement this CAIP.
+
+Below is an unverifed, non-normative, example for `wallet_getAssets` which is only supported in the `eip155` scope:
+
+```json
+"scopedProperties": {
+    "eip155": {
+        "caip345": [{
+            "methods": ["wallet_getAssets"],
+            "url": "https://wallet-service.example.com/rpc"
+        }],
+    }
+}
+```
+
+Below is an unverifed, non-normative, example for ERC-7836, which is also only valid in the `eip155` scope:
+
+```json
+"scopedProperties": {
+    "eip155": {
+        "caip345": [{
+            "methods": ["wallet_prepareCalls", "wallet_sendPreparedCalls"],
+            "url": "https://wallet-service.example.com/rpc"
+        }],
+    }
+}
+```
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+### Support for multiple wallet service URLs
+
+Supporting an array of wallet services increases flexibility. Allowing different URLs (servers or parameters) to be used for different methods or use cases.
+
+There must still be 1 canonical wallet service URL for a given method (if available at all).
+
+### Different wallet service depending on account
+
+There was consideration for being able to specify different wallet service URLs for different accounts. However, this would not make sense in the context of CAIP-25 because there is no mechanism to scope the methods to particular accounts. If this CAIP provided a way to scope the methods to particular accounts, the app may still try to send the method requests for non-listed accounts directly to the wallet.
+
+### Map methods to wallet services, instead of methods in array
+
+There was consideraiton for defining the standard to have a separate wallet service URL for every single method. However, this would cause excessive bandwidth consumption if the same URL were to be used for multiple methods which we think is the more likely case.
```

Comment:
```suggestion
There was consideration for defining the standard to have a separate wallet service URL for every single method. However, this would cause excessive bandwidth consumption if the same URL were to be used for multiple methods which we think is the more likely case.
```